handler.go

```diff
@@ -130,30 +133,31 @@ func (h *handler) fillByCachedUserContexts(contexts []*context.UserContext, reqU
 	return userContexts, nil
 }
 
-type optionsHandler struct {
-	methods []string
-}
+const ErrorMessageForMethodNotAllowed = "request resource does not support http method '%s'"
 
-func newOptionsHandler() *optionsHandler {
-	return &optionsHandler{
-		methods: make([]string, 0),
-	}
+func notAllowHandler(method string) router.Handler {
```

Comment:
I think `createNotAllowMethodHandler` is the better name, the current name like a variable, not a function and not sure it does not allow what.